src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3094,12 +3094,21 @@ private void AddSynthesizedRecordMembersIfNecessary(MembersAndInitializersBuilde
 
             return;
 
-            SynthesizedRecordConstructor addCtor(RecordDeclarationSyntax declWithParameters)
+            SynthesizedRecordConstructor? addCtor(RecordDeclarationSyntax declWithParameters)
             {
                 Debug.Assert(declWithParameters.ParameterList is object);
                 var ctor = new SynthesizedRecordConstructor(this, declWithParameters, diagnostics);
-                members.Add(ctor);
-                return ctor;
+
+                if (ctor.ParameterCount == 1 && ctor.Parameters[0].Type.Equals(this))
```

Comment:
>if (ctor.ParameterCount == 1 && ctor.Parameters[0].Type.Equals(this)) [](start = 16, length = 69)

Do we understand why a regular process of detecting ambiguous members doesn't detect the ambiguity